students/TinaB/lesson04/katafourteen.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+import random
+
+# book_text_file = "sherlock.txt"
+#"sherlock_small.txt"
+text_file = open("sherlock_small.txt", 'r')
+text = text_file.read()
+# first option takes text as is, second
+words_in_text = text.split()
+#words_in_text = text.replace('--', ' ').replace('.', '').replace(',', '').replace(
+#    '"', '').replace("'", '').replace(":", ' ').replace('?', '').replace('!', '').replace('\n', ' ').replace('(', '').replace(')', '').split()
+words_in_text = [str(i) for i in words_in_text]
+text_file.close()
+# print(words_in_text)
+
+
+def create_trigram(trigram_text):
+    '''Creates the trigram dictionary from text argument when called'''
+    trigram_dictionary = {}
+    # populates trigram by running a loop
+    for tri_word in range(0, len(trigram_text)-2):
+        # setdefault checks to see if key exists in dictionary and if not sets it with value of []
+        trigram_dictionary.setdefault(
+            '{} {}'.format(trigram_text[tri_word], trigram_text[tri_word+1]), [])
+
+        trigram_dictionary['{} {}'.format(trigram_text[tri_word], trigram_text[tri_word+1])
+                           ].append('{}'.format(trigram_text[tri_word+2]))
```

Comment:
You can chain the `append()` method to the `set default()` method on the previous line to make this a bit more concise